### PR TITLE
Add tests and method for list in describe_distribution

### LINKED ISSUE #97: Use describe_distribution() with a vector/list of variables
Currently, `describe_distribution()` works with the whole dataframe or with a single variable:
```r
datawizard::describe_distribution(mtcars)

#> Variable |   Mean |     SD |    IQR |  Min |   Max | Skewness | Kurtosis |  n | n_Missing
#> -----------------------------------------------------------------------------------------
#> mpg      |  20.09 |   6.03 |   7.53 | 10.4 |  33.9 |     0.67 |   -0.022 | 32 |         0
#> cyl      |   6.19 |   1.79 |   4.00 |  4.0 |   8.0 |    -0.19 |   -1.763 | 32 |         0
#> disp     | 230.72 | 123.94 | 221.53 | 71.1 | 472.0 |     0.42 |   -1.068 | 32 |         0
#> hp       | 146.69 |  68.56 |  84.50 | 52.0 | 335.0 |     0.80 |    0.275 | 32 |         0
#> drat     |   3.60 |   0.53 |   0.84 |  2.8 |   4.9 |     0.29 |   -0.450 | 32 |         0
#> wt       |   3.22 |   0.98 |   1.19 |  1.5 |   5.4 |     0.47 |    0.417 | 32 |         0
#> qsec     |  17.85 |   1.79 |   2.02 | 14.5 |  22.9 |     0.41 |    0.865 | 32 |         0
#> vs       |   0.44 |   0.50 |   1.00 |  0.0 |   1.0 |     0.26 |   -2.063 | 32 |         0
#> am       |   0.41 |   0.50 |   1.00 |  0.0 |   1.0 |     0.40 |   -1.967 | 32 |         0
#> gear     |   3.69 |   0.74 |   1.00 |  3.0 |   5.0 |     0.58 |   -0.895 | 32 |         0
#> carb     |   2.81 |   1.62 |   2.00 |  1.0 |   8.0 |     1.16 |    2.020 | 32 |         0

datawizard::describe_distribution(mtcars$mpg)

#> Mean | SD | IQR | Min | Max | Skewness | Kurtosis |  n | n_Missing
#> ------------------------------------------------------------------
#> 20   |  6 | 7.5 |  10 |  34 |     0.67 |   -0.022 | 32 |         0
```

It would be nice to be able to pass a custom list or vector of variables to it. Currently, if we provide a vector of variables, they are aggregated:
``` r
datawizard::describe_distribution(c(mtcars$mpg, mtcars$cyl))

#> Mean |  SD | IQR | Min | Max | Skewness | Kurtosis |  n | n_Missing
#> -------------------------------------------------------------------
#> 13   | 8.3 |  13 |   4 |  34 |     0.74 |    -0.41 | 64 |         0
```

<sup>Created on 2022-03-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### PR DESCRIPTION
**Pull request by etiennebacher**
This PR is related to #97. Sometimes we want to compare the distribution of several variables but not necessarily of the whole dataframe, so this PR adds a method to pass a list to `describe_distribution()`. Some examples below:
```r
> describe_distribution(list(mtcars$mpg, mtcars$cyl))

Variable   |  Mean |   SD |  IQR |          Range | Skewness | Kurtosis |  n | n_Missing
----------------------------------------------------------------------------------------
mtcars$mpg | 20.09 | 6.03 | 7.53 | [10.40, 33.90] |     0.67 |    -0.02 | 32 |         0
mtcars$cyl |  6.19 | 1.79 | 4.00 |   [4.00, 8.00] |    -0.19 |    -1.76 | 32 |         0

> describe_distribution(list(foo = mtcars$mpg, foo2 = mtcars$cyl))

Variable |  Mean |   SD |  IQR |          Range | Skewness | Kurtosis |  n | n_Missing
--------------------------------------------------------------------------------------
foo      | 20.09 | 6.03 | 7.53 | [10.40, 33.90] |     0.67 |    -0.02 | 32 |         0
foo2     |  6.19 | 1.79 | 4.00 |   [4.00, 8.00] |    -0.19 |    -1.76 | 32 |         0

> describe_distribution(list(foo = mtcars$mpg, mtcars$cyl))

Variable   |  Mean |   SD |  IQR |          Range | Skewness | Kurtosis |  n | n_Missing
----------------------------------------------------------------------------------------
foo        | 20.09 | 6.03 | 7.53 | [10.40, 33.90] |     0.67 |    -0.02 | 32 |         0
mtcars$cyl |  6.19 | 1.79 | 4.00 |   [4.00, 8.00] |    -0.19 |    -1.76 | 32 |         0

> x <- list(mtcars$mpg, mtcars$cyl)
> describe_distribution(x)

Variable |  Mean |   SD |  IQR |          Range | Skewness | Kurtosis |  n | n_Missing
--------------------------------------------------------------------------------------
Var_1    | 20.09 | 6.03 | 7.53 | [10.40, 33.90] |     0.67 |    -0.02 | 32 |         0
Var_2    |  6.19 | 1.79 | 4.00 |   [4.00, 8.00] |    -0.19 |    -1.76 | 32 |         0
```
Note that if the list is stored in an object (such as `x` above) and if its elements are unnamed, it is not possible to know the elements that created `x`, hence the "Var_1", "Var_2" as variable names. I also add tests and some doc for this.

There will probably be some things to change so I didn't add details in the NEWS yet. What do you think? 

Close #97.

### CODE REVIEW

File: R/describe_distribution.R
Comment (IndrajeetPatil) [inline R/describe_distribution.R:27]: Can you please format this to 80-character width?

File: R/describe_distribution.R
Comment (IndrajeetPatil) [inline R/describe_distribution.R:74]: For code readability, you should comment here the intent behind this complex conditional statement.

For example, 
> "The function currently doesn't support descriptive summaries for `character` or `factor` types."

File: R/describe_distribution.R
Comment (IndrajeetPatil) [inline R/describe_distribution.R:103]: Have you tried the plotting method in `{see}` and checked if it works with outputs from `list` input?

Maybe you can post an example in the PR thread.

**Review by IndrajeetPatil — CHANGES_REQUESTED** [review]
Looks good! Just a few minor comments that need to be attended to.

File: R/describe_distribution.R
Comment (etiennebacher) [reply R/describe_distribution.R:80]: I copied this part from `describe_distribution.dataframe()`, so it should be modified there also. One question about that is that `describe_distribution()` also support characters and factors. I think the output is slightly different for characters and factors, which is why it can't be aggregated in a single dataframe with numeric variables, but it should be possible to output two different tables *à la* `skimr`. What do you think?

File: R/describe_distribution.R
Comment (etiennebacher) [reply R/describe_distribution.R:109]: I didn't know there was a plotting method for that. I just tried `plot(describe_distribution(list(mtcars$mpg, mtcars$cyl)))` and it doesn't work. I don't have any experience with `ggplot2` programming so not sure how to handle this. I'll take a look at `see` source code later, maybe it's easier than I think

File: R/describe_distribution.R
Comment (IndrajeetPatil) [reply R/describe_distribution.R:80]: Comments should suffice for now.

Yeah, we are also not that happy with its behaviour with these types, but this should be handled in different PR.

See #46 and #45

File: R/describe_distribution.R
Comment (IndrajeetPatil) [reply R/describe_distribution.R:109]: I think you can leave this in for now and create an issue in `{see}` repository so that we don't forget about it.

**Review by IndrajeetPatil — APPROVED** [review]
Looks good to me! Thanks for working on this (and also for adding exhaustive unit tests!).

I will wait for @strengejacke to have a look at this before I merge.

File: R/describe_distribution.R
Comment (strengejacke) [inline R/describe_distribution.R:74]: I suggest using `.safe_deparse` instead of `deparse` here.

**Review by strengejacke — CHANGES_REQUESTED** [review]

File: R/describe_distribution.R
Comment (IndrajeetPatil) [reply R/describe_distribution.R:74]: @etiennebacher In case you wish to check out its definition, you can find it here:

https://github.com/easystats/datawizard/blob/2a4e53ba1f64c70386a4c8f704d4a46bbe6a81e0/R/demean.R#L415-L418

**Review by strengejacke — APPROVED** [review]

**Pull request merged**
